examples/multiagent/iql.py

```diff
@@ -0,0 +1,283 @@
+        scenario=scenario_name,
+        num_envs=config["evaluation_episodes"],
+        continuous_actions=False,
+        max_steps=max_steps,
+        device=vmas_device,
+        seed=seed,
+        # Scenario kwargs
+        **env_config,
+    )
+    env_config.update({"n_agents": env.n_agents, "scenario_name": scenario_name})
+
+    # Policy
+    net = MultiAgentMLP(
+        n_agent_inputs=env.observation_spec["agents", "observation"].shape[-1],
+        n_agent_outputs=env.action_spec.space.n,
+        n_agents=env.n_agents,
+        centralised=False,
+        share_params=model_config["shared_parameters"],
+        device=training_device,
+        depth=2,
+        num_cells=256,
+        activation_class=nn.Tanh,
+    )
+    module = TensorDictModule(
+        net, in_keys=[("agents", "observation")], out_keys=[("agents", "action_value")]
+    )
+    value_module = QValueModule(
+        action_value_key=("agents", "action_value"),
+        out_keys=[
+            env.action_key,
+            ("agents", "action_value"),
+            ("agents", "chosen_action_value"),
+        ],
+        spec=env.unbatched_action_spec,
+        action_space=None,
+    )
+    qnet = SafeSequential(module, value_module)
+
+    qnet_explore = EGreedyWrapper(
+        qnet,
+        eps_init=0.3,
+        eps_end=0,
+        annealing_num_steps=int(total_frames * (1 / 2)),
+        action_key=env.action_key,
+        spec=env.action_spec,
+    )
+
+    with set_exploration_type(ExplorationType.RANDOM):
+        qnet_explore(env.reset().to(training_device))
+
+    collector = SyncDataCollector(
+        env,
+        qnet_explore,
+        device=vmas_device,
+        storing_device=training_device,
+        frames_per_batch=frames_per_batch,
+        total_frames=total_frames,
+    )
+
+    replay_buffer = TensorDictReplayBuffer(
+        storage=LazyTensorStorage(memory_size, device=training_device),
+        sampler=SamplerWithoutReplacement(),
+        batch_size=config["minibatch_size"],
+    )
+
+    loss_module = DQNLoss(qnet, delay_value=True)
+    loss_module.set_keys(
+        action_value=("agents", "action_value"),
+        action=env.action_key,
+        value=("agents", "chosen_action_value"),
+        reward=env.reward_key,
+    )
+    loss_module.make_value_estimator(ValueEstimators.TD0, gamma=config["gamma"])
+    target_net_updater = SoftUpdate(loss_module, eps=1 - config["tau"])
+
+    optim = torch.optim.Adam(loss_module.parameters(), config["lr"])
+
+    # Logging
+    if log:
+        config.update({"model": model_config, "env": env_config})
+        model_name = ("Het" if not model_config["shared_parameters"] else "") + "IQL"
+        logger = WandbLogger(
+            exp_name=generate_exp_name(env_config["scenario_name"], model_name),
+            project=f"torchrl_{env_config['scenario_name']}",
+            group=model_name,
+            save_code=True,
+            config=config,
+        )
+        wandb.run.log_code(".")
```

Comment:
here too, i'd use a hydra structured config where we pass `config.logger` to `make_logger`. If `make_logger` returns `None` then it means that we're not logging. You can check other single agent algos for examples